core/trino-main/src/main/java/io/trino/execution/scheduler/FallbackToFullNodePartitionMemoryEstimator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler;
+
+import io.airlift.units.DataSize;
+import io.trino.Session;
+import io.trino.spi.ErrorCode;
+import io.trino.spi.StandardErrorCode;
+import io.trino.spi.TrinoException;
+
+import static io.trino.SystemSessionProperties.getQueryMaxTotalMemoryPerTask;
+import static io.trino.spi.StandardErrorCode.EXCEEDED_LOCAL_MEMORY_LIMIT;
+
+public class FallbackToFullNodePartitionMemoryEstimator
+        implements PartitionMemoryEstimator
+{
+    // temporarily express full-node requirement as huge amount of memory
+    public static final DataSize FULL_NODE_MEMORY = DataSize.of(512, DataSize.Unit.GIGABYTE);
+
+    private static final MemoryRequirements FULL_NODE_MEMORY_REQUIREMENTS = new MemoryRequirements(FULL_NODE_MEMORY, true);
+
+    @Override
+    public MemoryRequirements getInitialMemoryRequirements(Session session)
+    {
+        return new MemoryRequirements(
+                getQueryMaxTotalMemoryPerTask(session).orElseThrow(() -> new TrinoException(StandardErrorCode.CONFIGURATION_INVALID, "task memory limit not set")),
```

Comment:
Not convinced. It feels to me that description which says it is used for only when task level retries are enabled is enough. But let me know if you feel strongly.

Reply:
I was thinking about aligning the name with these two properties: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/SystemSessionProperties.java#L151

Because the value set for the memory limit is tightly related to values chosen for the `FAULT_TOLERANT_EXECUTION_TARGET_TASK_INPUT_SIZE` and `FAULT_TOLERANT_EXECUTION_TARGET_TASK_SPLIT_COUNT`

Reply:
renamed